base/markdown/parse/parse.jl

```diff
@@ -51,7 +51,9 @@ function parseinline(stream::IO, md::MD, config::Config)
     content = []
     buffer = IOBuffer()
     while !eof(stream)
-        char = peek(stream)
+        # FIXME: this is broken if we're looking for non-ASCII
```

Comment:
we need `peek` to match the current `read` interface:
```julia
read(io, UInt8)
read(io, Char)

peek(io, UInt8)
peek(io, Char)
```
with the defaults matching.